clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -96,15 +151,20 @@ public FetchRequest(int replicaId, int maxWait, int minBytes, Map<TopicPartition
         this.replicaId = replicaId;
         this.maxWait = maxWait;
         this.minBytes = minBytes;
-        this.fetchData = fetchData;
+        this.maxBytes = maxBytes;
+        this.fetchData = new LinkedHashMap<>(fetchData);
```

Comment:
I didn't mean to have that, it is a relic from when the code was taking a plain `Map`. So I am inclined to remove this copy unless you think we should have it?